src/passes/GenerateDynCalls.cpp

```diff
@@ -36,21 +36,103 @@ namespace wasm {
 struct GenerateDynCalls : public WalkerPass<PostWalker<GenerateDynCalls>> {
   GenerateDynCalls(bool onlyI64) : onlyI64(onlyI64) {}
 
+  void doWalkModule(Module* wasm) {
+    PostWalker<GenerateDynCalls>::doWalkModule(wasm);
+    for (auto& sig : invokeSigs) {
+      generateDynCallThunk(sig);
+    }
+  }
+
   void visitTable(Table* table) {
+    // Generate dynCalls for functions in the table
     if (table->segments.size() > 0) {
-      EmscriptenGlueGenerator generator(*getModule());
-      generator.onlyI64DynCalls = onlyI64;
       std::vector<Name> tableSegmentData;
       for (const auto& indirectFunc : table->segments[0].data) {
-        generator.generateDynCallThunk(
-          getModule()->getFunction(indirectFunc)->sig);
+        generateDynCallThunk(getModule()->getFunction(indirectFunc)->sig);
       }
     }
   }
 
+  void visitFunction(Function* func) {
+    // Generate dynCalls for invokes
+    if (func->imported() && func->base.startsWith("invoke_")) {
```

Comment:
please check also that `func->module == ENV`

Reply:
Done